### PR TITLE
Color Management- added color management support for simple expected files on Deadline

### PR DESCRIPTION
**Pull request by kalisp**
## Changelog Description
Running of `ExtractOIIOTranscode` during Deadline publish was previously implemented only on DCCs with AOVs (Maya, Max).
This PR extends this for other DCCs with flat structure of expected files.

## Testing notes:
1. Configure Color Management for your DCC of your choice
2. Publish to DL, check if representation were properly color modified

### CODE REVIEW

**Comment by jakubjezek001** [comment]
Task linked: [OP-5864 cleanup_farm: index out of range](https://app.clickup.com/t/6658547/OP-5864)

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py [hunk 1]
```diff
@@ -733,6 +735,13 @@ def _get_representations(self, instance, exp_files, do_not_add_review):
             if not already_there:
                 representations.append(rep)
 
+        for rep in representations:
+            # inject colorspace data
+            self.set_representation_colorspace(
+                rep, self.context,
+                colorspace=instance["colorspace"]
```
Comment (BigRoy) [inline openpype/modules/deadline/plugins/publish/submit_publish_job.py:742]: Typo?
```suggestion
                colorspace=instance.data["colorspace"]
```

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py [hunk 2]
```diff
@@ -861,7 +870,8 @@ def process(self, instance):
             "jobBatchName": data.get("jobBatchName", ""),
             "useSequenceForReview": data.get("useSequenceForReview", True),
             # map inputVersions `ObjectId` -> `str` so json supports it
-            "inputVersions": list(map(str, data.get("inputVersions", [])))
+            "inputVersions": list(map(str, data.get("inputVersions", []))),
+            "colorspace": instance.data['colorspace']
```
Comment (BigRoy) [inline openpype/modules/deadline/plugins/publish/submit_publish_job.py:874]: It seems pretty bold that `colorspace` key exists in `instance.data`. Or is there a plugin that's ensuring that key is there? :) I haven't seen that before in `instance.data` to begin with.

Comment (kalisp) [reply openpype/modules/deadline/plugins/publish/submit_publish_job.py]: Same say "bold", same say "stupid" :D.

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py (hunk 1 above)
Comment (kalisp) [reply openpype/modules/deadline/plugins/publish/submit_publish_job.py]: It is actually `instance.data`, renamed to make it clearer. Got similar issue previously too.

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py [hunk 3]
```diff
@@ -733,6 +736,13 @@ def _get_representations(self, instance, exp_files, do_not_add_review):
             if not already_there:
                 representations.append(rep)
 
+        for rep in representations:
+            # inject colorspace data
+            self.set_representation_colorspace(
+                rep, self.context,
+                colorspace=instance_data["colorspace"]
```
Comment (Minkiu) [inline openpype/modules/deadline/plugins/publish/submit_publish_job.py:743]: Wouldn't it be safer to do `instance_data.get` and default to something?

Comment (jakubjezek001) [reply openpype/modules/deadline/plugins/publish/submit_publish_job.py:743]: Agreed - please lets make this optional.

**Review by jakubjezek001 — APPROVED** [review]
LGTM

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py (hunk 3 above)
Comment (kalisp) [reply openpype/modules/deadline/plugins/publish/submit_publish_job.py:743]: It is filled on ln 875, so the key will always be there. 
`set_representation_colorspace` handles case where colorspace is None, in that case it uses File rules and `get_imageio_colorspace_from_filepath` to get some value, so I don't think it is necessary to provide any default value here.

**Pull request merged**